File: app/src/main/java/com/loopeer/codereader/ui/view/AddRepoChecker.java
package com.loopeer.codereader.ui.view;

import android.text.TextUtils;

public class AddRepoChecker extends Checker {
    public String repoName;
    public String repoDownloadUrl;

    public AddRepoChecker(CheckObserver checkObserver) {
        super(checkObserver);
    }

    public void setRepoName(String repoName) {
        this.repoName = repoName;
        mCheckObserver.check(isEnable());
    }

    public void setRepoDownloadUrl(String repoDownloadUrl) {
        this.repoDownloadUrl = repoDownloadUrl;
        mCheckObserver.check(isEnable());
    }

    @Override
    public boolean isEnable() {
        return  !TextUtils.isEmpty(repoDownloadUrl);
    }
}

File: code-reader-kt/src/main/java/com/loopeer/codereaderkt/ui/view/AddRepoChecker.kt
package com.loopeer.codereaderkt.ui.view

import android.text.TextUtils

class AddRepoChecker(checkObserver: Checker.CheckObserver) : Checker(checkObserver) {
    override val isEnable: Boolean
        get() = !TextUtils.isEmpty(repoDownloadUrl) //To change initializer of created properties use File | Settings | File Templates.

    var repoName:String?=null
    set(value) {
        field = value
        mCheckObserver.check(isEnable)
    }
    var repoDownloadUrl: String?=null
    set(value) {
        field = value
        mCheckObserver.check(isEnable)
    }
}
